### PR TITLE
tests: add core revert test

### PR DESCRIPTION
**Pull request by fgimenez**
This test exercises a core revert on a nested ubuntu-core system, currently only a preinstalled snap is set (network-manager) and the check after the revert only takes into account network configuration. It can be executed with: `SPREAD_CORE_CHANNEL=stable spread -v qemu:ubuntu-16.04-64:tests/nested/core-revert`.

This PR also includes a refactor of reusable functions from a previous nested task, the actual test is https://github.com/snapcore/snapd/pull/3348/files#diff-4825fd024a235e7895b109ad37072e68

### CODE REVIEW

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
+
+    # create ubuntu-core image
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
+    trap 'snap remove ubuntu-image' EXIT
```
Comment (mvo5) [inline tests/lib/nested.sh]: Trap will replace other existing traps on the same signal. So if we alread use a `trap ... EXIT` anwhere this is overriden now. We may consider switching to a helper (`add_trap`) that stacks the cleanups if we already use traps. Probably material for a new PR though.

Comment (fgimenez) [inline tests/lib/nested.sh]: Thanks @mvo5 ! Indeed, for the time being, and given that all the tasks in the nested suite are going to use ubuntu-image, I've moved its install/unistall at the suite level, and btw also removed a trap used for apt, please take another look.

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+create_nested_core_vm(){
+    # determine arch related vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
```
Comment (fgimenez) [inline tests/lib/nested.sh]: Sure, will do in a followup

File: tests/lib/nested.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
+
+    # create ubuntu-core image
+    mkdir -p /tmp/work-dir
```
Comment (fgimenez) [inline tests/lib/nested.sh]: Ok, noted for the followup

File: tests/nested/core-revert/task.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+summary: core revert test
```
Comment (fgimenez) [inline tests/nested/core-revert/task.yaml]: Sure, I'll add a details section

File: tests/nested/core-revert/task.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+systems: [ubuntu-16.04-64]
```
Comment (fgimenez) [inline tests/nested/core-revert/task.yaml]: I think so, in fact the interesting part happens on the nested machine, if we can install qemu all should be good. I'll try locally with debian/fedora anyway.

**Pull request merged**